src/NixFromNpm/Npm/Types.hs

```diff
@@ -25,14 +25,72 @@ data PackageInfo = PackageInfo {
   piTags :: Record SemVer
   } deriving (Show, Eq)
 
+-- | Taken from https://nodejs.org/api/process.html#process_process_platform,
+-- and filtered to those that correspond to nixpkgs platforms.
+data NodePlatform
+  = Darwin
+  | FreeBSD
+  | OpenBSD
+  | Linux
+  | SunOS
+  deriving (Show, Eq)
+
+-- | Convert a NodePlatform into text
+nodePlatformToText :: IsString t => NodePlatform -> t
+nodePlatformToText = \case
+  Darwin -> "darwin"
+  FreeBSD -> "freebsd"
+  OpenBSD -> "openbsd"
+  Linux -> "linux"
+  SunOS -> "solaris"
+
+-- | Parse a node platform from a string.
+parseNodePlatform :: Alternative f => Text -> f NodePlatform
+parseNodePlatform = \case
+  "linux" -> pure Linux
+  "darwin" -> pure Darwin
+  "freebsd" -> pure FreeBSD
+  "openbsd" -> pure OpenBSD
+  "sunos" -> pure SunOS
+  _ -> empty
+
 -- | Metadata about a package.
 data PackageMeta = PackageMeta {
   pmDescription :: Maybe Text,
   pmAuthor :: Maybe Text,
   pmHomepage :: Maybe URI,
-  pmKeywords :: Vector Text
+  pmKeywords :: Vector Text,
+  pmPlatforms :: Vector NodePlatform
   } deriving (Show, Eq)
 
+-- | Default (empty) package metadata.
+emptyPackageMeta :: PackageMeta
+emptyPackageMeta = PackageMeta Nothing Nothing Nothing mempty mempty
+
+instance FromJSON PackageMeta where
```

Comment:
I moved the code for this out of the `VersionInfo` instance and into its own instance, which seems a bit more modular/clear :)